enclave/core/sgx/thread.c

```diff
@@ -242,8 +325,18 @@ oe_result_t oe_mutex_lock(oe_mutex_t* mutex)
         oe_spin_lock(&m->lock);
         {
             /* Attempt to acquire lock */
-            if (_mutex_lock(m, self) == 0)
+            oe_result_t result = _mutex_lock(m, self);
+
+            if (result == OE_BUSY)
+            {
+                /* Deadlock */
+                OE_TRACE_ERROR("Recursively acquire a non-recursive lock");
```

Comment:
Would this print the trace message in each iteration?

Reply:
No, the execution will proceed to call the oe_spin_lock above again and hang.